azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -770,7 +784,7 @@ public Status getImpliedStatus(ExecutableNode node) {
       ExecutableNode dependencyNode = flow.getExecutableNode(dependency);
       Status depStatus = dependencyNode.getStatus();
 
-      if (!Status.isStatusFinished(depStatus)) {
+      if (!Status.isStatusFinished(depStatus) || (depStatus == Status.FAILED && isInRetry(dependencyNode))) {
```

Comment:
I am not sure why the node status will be failed if it is being retried.
azkaban.execapp.FlowRunner#retryJobIfPossible should have set the status to READY again.

Reply:
Based on the example you gave in the issue #862, here is my interpretation of what may have happened, please correct me if I am wrong.

JobB finished successfully first, it is added to the finishedNodes.  azkaban.execapp.FlowRunner#progressGraph adds JobC to the nodesToCheck list. azkaban.execapp.FlowRunner#runReadyJob calls azkaban.execapp.FlowRunner#getImpliedStatus, at this time, jobA fails it sets its status to failed in azkaban.execapp.JobRunner#runJob. But it has yet to have a chance to be added to the finishedNodes list so that retryJobIfPossible has not been called to reset the status. JobC is canceld because it sees that jobA has failed.

Reply:
@HappyRay  Sorry for the delay in repose. 
Yes. Your understanding is correct. Ideally the downstream jobs should only be cancelled after all the retries are exhausted.

Reply:
Thanks. Could you update the change description to include more details such as what I described above?